File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> sc;
        int result=0;

        for (char chr : s){
            sc[chr]++;
        }

        for (auto& pair : sc) {
            result+= (pair.second / 2) * 2;
            if (result % 2 == 0 && pair.second % 2 == 1) {
                result+=1;
            }
        }
        return result;
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.go
func longestPalindrome(s string) int {
    sc := make(map[rune]int)
    result := 0

    for _, char := range s {
        sc[char]++
    }

    for _, val := range sc {
        result += (val / 2) * 2
        if result%2==0 && val%2==1 {
            result++
        }
    }
    return result    
}